internal/server/server.go

```diff
@@ -0,0 +1,118 @@
+package server
+
+import (
+	"html/template"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/sirius"
+)
+
+type Logger interface {
+	Request(*http.Request, error)
+}
+
+type Server struct {
+	Templates map[string]*template.Template
+	Client    *sirius.Client
+}
+
+func getContext(r *http.Request) sirius.Context {
+	token := ""
+
+	if r.Method == http.MethodGet {
+		if cookie, err := r.Cookie("XSRF-TOKEN"); err == nil {
+			token, _ = url.QueryUnescape(cookie.Value)
+		}
+	} else {
+		token = r.FormValue("xsrfToken")
+	}
+
+	return sirius.Context{
+		Context:   r.Context(),
+		Cookies:   r.Cookies(),
+		XSRFToken: token,
+	}
+}
+
+type Template interface {
+	ExecuteTemplate(io.Writer, string, interface{}) error
+}
+
+type Client interface {
+	WarningClient
+}
+
+func New(logger Logger, client Client, templates map[string]*template.Template, prefix, siriusPublicURL, webDir string) http.Handler {
+	wrap := errorHandler(logger, templates["error.gohtml"], prefix, siriusPublicURL)
+
+	mux := http.NewServeMux()
+
+	mux.Handle("/", http.NotFoundHandler())
+	mux.HandleFunc("/health-check", func(w http.ResponseWriter, r *http.Request) {})
+	mux.Handle("/create-warning", wrap(Warning(client, templates["warning.gohtml"])))
+
+	static := http.FileServer(http.Dir("web/static"))
+	mux.Handle("/assets/", static)
+	mux.Handle("/javascript/", static)
+	mux.Handle("/stylesheets/", static)
+
+	return http.StripPrefix(prefix, securityHeaders(mux))
+}
+
+func securityHeaders(h http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add("Content-Security-Policy", "default-src 'self'")
+		w.Header().Add("Referrer-Policy", "same-origin")
+		w.Header().Add("Strict-Transport-Security", "max-age=31536000; includeSubDomains; preload")
+		w.Header().Add("X-Content-Type-Options", "nosniff")
+		w.Header().Add("X-Frame-Options", "SAMEORIGIN")
+		w.Header().Add("X-XSS-Protection", "1; mode=block")
+
+		h.ServeHTTP(w, r)
+	}
+}
+
+type Handler func(w http.ResponseWriter, r *http.Request) error
+
+type errorVars struct {
+	SiriusURL string
+	Path      string
+	Code      int
+	Error     string
+}
+
+type unauthorizedError interface {
+	IsUnauthorized() bool
+}
```

Comment:
I think this is cleaner than having `ErrUnauthorized` as every status related problem can just be a `StatusError`, and we then implement this method to force a logout